### PR TITLE
Features

### PR DESCRIPTION
**Pull request by ChanyutJean**

### CODE REVIEW

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+	@Autowired
+	private EventRegistrationRepository eventRegistrationRepository;
+
+	@Override
+	@Transactional
+	public void createEvent(CreateEventRequest request, StreamObserver<Result> responseObserver) {
+
+		if (eventRepository.findById(request.getEvent().getId()).isPresent()) {
+
+			Result result = ServiceUtil.returnError("An event with this ID already exists.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Should we custom error as new class?

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+		}
+
+		Result result = ServiceUtil.returnSuccessful("Event deletion successful.");
+
+		ServiceUtil.configureResponseObserver(responseObserver, result);
+	}
+
+	@Override
+	@Transactional
+	public void updateEventFacility(UpdateEventFacilityRequest request, StreamObserver<Result> responseObserver) {
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Why need this since event only can add facility via facility request?

Comment (ChanyutJean) [reply src/main/java/app/onepass/organizer/services/EventService.java]: for when organizers want to change facility for an event

Comment (new5558) [reply src/main/java/app/onepass/organizer/services/EventService.java]: But each facility need to be change though facility request?

Comment (new5558) [reply src/main/java/app/onepass/organizer/services/EventService.java]: I can create method for you in facility service

File: src/main/java/app/onepass/organizer/services/EventService.java (hunk 1 above)
Comment (ChanyutJean) [reply src/main/java/app/onepass/organizer/services/EventService.java]: return error is not an error class, it’s a method for returning an error result of isok = false

File: src/main/java/app/onepass/organizer/utilities/DatabaseExceptionCatcher.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	public static <M extends GeneratedMessageV3> void
+	catcher(BiConsumer<M, StreamObserver<Result>> consumer, M request, StreamObserver<Result> responseObserver) {
+
+		try {
+
+			consumer.accept(request, responseObserver);
+
+		} catch (DataAccessException exception) {
+
+			Result result = ServiceUtil.returnError("Unable to manipulate database.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/utilities/DatabaseExceptionCatcher.java]: Why no grpc error code here?

File: src/main/java/app/onepass/organizer/services/TagService.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	@Autowired
+	EventTagRepository eventTagRepository;
+
+	@Override
+	@Transactional
+	public void createTag(CreateTagRequest request, StreamObserver<Result> responseObserver) {
+
+		if (tagRepository.findById(request.getTag().getId()).isPresent()) {
+
+			Result result = ServiceUtil.returnError("A tag with this ID already exists.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/TagService.java]: Error code

File: src/main/java/app/onepass/organizer/services/OrganizationService.java [hunk 5]
```diff
@@ -81,30 +98,26 @@ public void readOrganizationById(ReadByIdRequest request, StreamObserver<ReadOrg
...
+        ServiceUtil.configureResponseObserver(responseObserver, getOrganizationByIdResult);
 
     }
 
     @Override
+    @Transactional
     public void updateOrganization(UpdateOrganizationRequest request, StreamObserver<Result> responseObserver) {
 
-        long organizationId = request.getOrganizationId();
+        if (!organizationRepository.findById(request.getOrganization().getId()).isPresent()) {
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/OrganizationService.java:108]: Error code

File: src/main/java/app/onepass/organizer/services/OrganizationService.java [hunk 6]
```diff
@@ -1,36 +1,54 @@
...
+    @Autowired
+    private UserOrganizationRepository userOrganizationRepository;
+
     @Override
+    @Transactional
     public void createOrganization(CreateOrganizationRequest request, StreamObserver<Result> responseObserver) {
 
+        if (organizationRepository.findById(request.getOrganization().getId()).isPresent()) {
+
+            Result result = ServiceUtil.returnError("An organization with this ID already exists.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/OrganizationService.java]: Error code

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+
+		Result result;
+
+		if (organizationId == request.getOrganizationId()) {
+
+			result = ServiceUtil.returnSuccessful("The specified organization has the specified event.");
+
+		} else {
+
+			result = ServiceUtil.returnSuccessful("The event is not hosted by this organization");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Return error

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+
+		try {
+
+			eventEntity = eventRepository
+					.findById(request.getEventId())
+					.orElseThrow(IllegalArgumentException::new);
+
+		} catch (IllegalArgumentException illegalArgumentException) {
+
+			Result result = ServiceUtil.returnSuccessful("The event does not exist.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Return error

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+		ServiceUtil.configureResponseObserver(responseObserver, result);
+	}
+
+	@Override
+	@Transactional
+	public void updateEventFacility(UpdateEventFacilityRequest request, StreamObserver<Result> responseObserver) {
+
+		if (!facilityRepository.findById(request.getFacility().getId()).isPresent()) {
+
+			Result result = ServiceUtil.returnError("A facility with this ID does not exist.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Error code

File: src/main/java/app/onepass/organizer/services/EventService.java [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+	@Transactional
+	public void removeEvent(RemoveEventRequest request, StreamObserver<Result> responseObserver) {
+
+		long eventId = request.getEventId();
+
+		boolean deleteSuccessful = ServiceUtil.deleteEntity(eventId, eventRepository);
+
+		if (!deleteSuccessful) {
+
+			Result result = ServiceUtil.returnError("Cannot find event from given ID.");
```
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Error code

File: src/main/java/app/onepass/organizer/services/EventService.java (hunk 1 above)
Comment (new5558) [inline src/main/java/app/onepass/organizer/services/EventService.java]: Error code

**Review by new5558 — CHANGES_REQUESTED** [review]

**Comment by ChanyutJean** [comment]
@new5558 The following code was created before error codes are agreed among the team, I will be fixing it so please do not flood the board with comments.

**Comment by ChanyutJean** [comment]
@new5558 Please approve changes you requested.

**Review by new5558 — APPROVED** [review]

**Pull request merged**